.github/workflows/ci.yml

```diff
@@ -35,11 +33,7 @@ jobs:
       - name: Coverage
         run: |
           make -C test/unit-test/build/ coverage
-          lcov --rc lcov_branch_coverage=1 --remove test/unit-test/build/coverage.info '*test*' --output-file test/unit-test/build/coverage.info
```

Comment:
not resetting the coverage with each run, will give erroneous results when making changes
and it adds coverage with each run.
so if you change something in the code, that results in no longer covering a certain line..it will still show as covered while it is actually not.
coverage.info needs to be zeroed/removed with each run